### ISSUE TITLE
Preview not showing up after error

### ISSUE THREAD

**Issue by nairaner** [openscad/openscad#1727]
Everything works correctly until there is an issue in the code, then it stops showing preview, but it is still being rendered as you can see performance hit when rotating viewport. Happens with both Goldfeather and without it.
Happens if file have includes.

**Comment by t-paul** [comment]
Please give some more details as mentioned in "[How do I report bugs](https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/FAQ#How_do_I_report_bugs.3F)".

**Comment by nairaner** [comment]
I'm using latest git version on Archliux.
OpenGL 3.3
```
include <MCAD/stepper.scad>;

module plate(nemaStandard = Nema17)
{
    motorDiameter=nemaStandard[NemaSideSize][1];
    
    screwDistance=(nemaStandard[NemaSideSize][1]-
    nemaStandard[NemaDistanceBetweenMountingHoles][1])/2;
    
    difference()
    {
        minkowski()
        {
            cube([motorDiameter-6,motorDiameter-6,2]);
            cylinder(2,3,3);
        }
    translate([screwDistance-3,screwDistance-3,-0.1])
        cylinder(4.2,2,2);
    translate([motorDiameter-screwDistance-3,screwDistance-3,-0.1])
        cylinder(4.2,2,2);
    translate([screwDistance-3,motorDiameter-screwDistance-3,-0.1])
        cylinder(4.2,2,2);
    translate([motorDiameter-screwDistance-3,motorDiameter-screwDistance-3,-0.1])
        cylinder(4.2,2,2);
    translate([motorDiameter/2-3,motorDiameter/2-3,-0.1])
        cylinder(h=4.2,r=nemaStandard[NemaRoundExtrusionDiameter][1]/2+1);
    }
}
plate();
```

**Comment by MichaelAtOz** [comment]
`until there is an issue in the code` What do you mean by that? If there is an issue, i.e. bug, then it doesn't have anything to preview.
The code above works ok for me.

**Comment by nairaner** [comment]
@MichaelAtOz I mean if there is any kind of the problem with compiling, then after fixing it preview dont show anything.
https://www.youtube.com/watch?v=N4JPlzeSYgk

**Comment by MichaelAtOz** [comment]
The video shows v 20160630, later than the Development snapshot.
I'm on windows anyway, but the latest snapshot 20160608 does not have the issue.
Needs others on *nix to test.

**Comment by nairaner** [comment]
Like i said i am on latest git version. I was able to narrow down this bug a little bit.

```
module part()
{
    translate([0,0,5])
    cube(4);
    mirror([0,0,1])
    translate([0,0,5])
    cube(4);    
    cylinder(r=4,h=10,center=true);
}

difference()
{
    part();
    cube(1); //change this
}
```

**Comment by t-paul** [comment]
Nice, trimmed down examples help a lot. I think I saw this effect too, but could not really pinpoint the cause. I'll have a look if that example code triggers that for me too...

**Comment by nairaner** [comment]
Also it seems like F12 preview works, but not F9.

**Comment by nairaner** [comment]
Bug gets also triggered by resizing the preview, here's the example:
https://www.youtube.com/watch?v=gcUGOHI3SNc

**Comment by t-paul** [comment]
Could you add the text from "Help -> Library Info" to see what library versions are used? I wonder if that issue got dragged in by a Qt upgrade.

Would you be able to test this with a manually built version against Qt4?

**Comment by nairaner** [comment]
It doesn't happen on Qt4.
```
OpenSCAD Version: 20160630
System information: Linux 4.6.3-1-ARCH #1 SMP PREEMPT Fri Jun 24 21:19:13 CEST 2016 x86_64 Antergos Linux 4 CPUs 7.49 GB RAM
Compiler: GCC "6.1.1 20160602" 64bit
Boost version: 1_60
Eigen version: 3.2.8
CGAL version, kernels: 4.8, Cartesian, Extended_cartesian, Epeck
OpenCSG version: OpenCSG 1.4.0
Qt version: 5.7.0
QScintilla version: 2.9.2
MingW build: No
GLib version: 2.48.1
Application Path: /usr/bin
Documents Path: /home/nairaner/.local/share
Resource Path: /usr/share/openscad
User Library Path: /home/nairaner/.local/share/OpenSCAD/libraries
User Config Path: /home/nairaner/.config/OpenSCAD
Backup Path: /home/nairaner/.local/share/OpenSCAD/backups
OPENSCADPATH: 
OpenSCAD library path:
/home/nairaner/.local/share/OpenSCAD/libraries
/usr/share/openscad/libraries

OPENSCAD_FONT_PATH: 
OpenSCAD font path:
GLEW version: 1.13.0
OpenGL Version: 3.0 Mesa 11.2.2
GL Renderer: Mesa DRI Intel(R) Ivybridge Mobile 
GL Vendor: Intel Open Source Technology Center
RGBA(8880), depth(0), stencil(0)
GL_ARB_framebuffer_object: yes
GL_EXT_framebuffer_object: yes
GL_EXT_packed_depth_stencil: yes

Using QOpenGLWidget

GL Extensions:
GL_3DFX_texture_compression_FXT1
GL_AMD_conservative_depth
GL_AMD_draw_buffers_blend
GL_AMD_performance_monitor
GL_AMD_seamless_cubemap_per_texture
GL_AMD_shader_trinary_minmax
GL_ANGLE_texture_compression_dxt3
GL_ANGLE_texture_compression_dxt5
GL_APPLE_object_purgeable
GL_APPLE_packed_pixels
GL_APPLE_vertex_array_object
GL_ARB_ES2_compatibility
GL_ARB_ES3_compatibility
GL_ARB_arrays_of_arrays
GL_ARB_blend_func_extended
GL_ARB_buffer_storage
GL_ARB_clear_buffer_object
GL_ARB_clear_texture
GL_ARB_clip_control
GL_ARB_color_buffer_float
GL_ARB_compressed_texture_pixel_storage
GL_ARB_conditional_render_inverted
GL_ARB_conservative_depth
GL_ARB_copy_buffer
GL_ARB_copy_image
GL_ARB_debug_output
GL_ARB_depth_buffer_float
GL_ARB_depth_clamp
GL_ARB_depth_texture
GL_ARB_derivative_control
GL_ARB_draw_buffers
GL_ARB_draw_buffers_blend
GL_ARB_draw_elements_base_vertex
GL_ARB_draw_instanced
GL_ARB_explicit_attrib_location
GL_ARB_explicit_uniform_location
GL_ARB_fragment_coord_conventions
GL_ARB_fragment_program
GL_ARB_fragment_program_shadow
GL_ARB_fragment_shader
GL_ARB_framebuffer_no_attachments
GL_ARB_framebuffer_object
GL_ARB_framebuffer_sRGB
GL_ARB_get_program_binary
GL_ARB_get_texture_sub_image
GL_ARB_half_float_pixel
GL_ARB_half_float_vertex
GL_ARB_instanced_arrays
GL_ARB_internalformat_query
GL_ARB_invalidate_subdata
GL_ARB_map_buffer_alignment
GL_ARB_map_buffer_range
GL_ARB_multi_bind
GL_ARB_multisample
GL_ARB_multitexture
GL_ARB_occlusion_query
GL_ARB_occlusion_query2
GL_ARB_pipeline_statistics_query
GL_ARB_pixel_buffer_object
GL_ARB_point_parameters
GL_ARB_point_sprite
GL_ARB_program_interface_query
GL_ARB_provoking_vertex
GL_ARB_robustness
GL_ARB_sample_shading
GL_ARB_sampler_objects
GL_ARB_seamless_cube_map
GL_ARB_seamless_cubemap_per_texture
GL_ARB_separate_shader_objects
GL_ARB_shader_atomic_counters
GL_ARB_shader_bit_encoding
GL_ARB_shader_clock
GL_ARB_shader_draw_parameters
GL_ARB_shader_image_load_store
GL_ARB_shader_image_size
GL_ARB_shader_objects
GL_ARB_shader_storage_buffer_object
GL_ARB_shader_texture_image_samples
GL_ARB_shader_texture_lod
GL_ARB_shading_language_100
GL_ARB_shading_language_420pack
GL_ARB_shading_language_packing
GL_ARB_shadow
GL_ARB_sync
GL_ARB_texture_barrier
GL_ARB_texture_border_clamp
GL_ARB_texture_compression
GL_ARB_texture_compression_bptc
GL_ARB_texture_compression_rgtc
GL_ARB_texture_cube_map
GL_ARB_texture_cube_map_array
GL_ARB_texture_env_add
GL_ARB_texture_env_combine
GL_ARB_texture_env_crossbar
GL_ARB_texture_env_dot3
GL_ARB_texture_float
GL_ARB_texture_gather
GL_ARB_texture_mirror_clamp_to_edge
GL_ARB_texture_mirrored_repeat
GL_ARB_texture_multisample
GL_ARB_texture_non_power_of_two
GL_ARB_texture_query_levels
GL_ARB_texture_query_lod
GL_ARB_texture_rectangle
GL_ARB_texture_rg
GL_ARB_texture_rgb10_a2ui
GL_ARB_texture_storage
GL_ARB_texture_storage_multisample
GL_ARB_texture_swizzle
GL_ARB_texture_view
GL_ARB_timer_query
GL_ARB_transform_feedback2
GL_ARB_transform_feedback3
GL_ARB_transform_feedback_instanced
GL_ARB_transpose_matrix
GL_ARB_uniform_buffer_object
GL_ARB_vertex_array_bgra
GL_ARB_vertex_array_object
GL_ARB_vertex_attrib_binding
GL_ARB_vertex_buffer_object
GL_ARB_vertex_program
GL_ARB_vertex_shader
GL_ARB_vertex_type_10f_11f_11f_rev
GL_ARB_vertex_type_2_10_10_10_rev
GL_ARB_window_pos
GL_ATI_blend_equation_separate
GL_ATI_draw_buffers
GL_ATI_separate_stencil
GL_ATI_texture_env_combine3
GL_ATI_texture_float
GL_EXT_abgr
GL_EXT_bgra
GL_EXT_blend_color
GL_EXT_blend_equation_separate
GL_EXT_blend_func_separate
GL_EXT_blend_minmax
GL_EXT_blend_subtract
GL_EXT_compiled_vertex_array
GL_EXT_copy_texture
GL_EXT_draw_buffers2
GL_EXT_draw_instanced
GL_EXT_draw_range_elements
GL_EXT_fog_coord
GL_EXT_framebuffer_blit
GL_EXT_framebuffer_multisample
GL_EXT_framebuffer_multisample_blit_scaled
GL_EXT_framebuffer_object
GL_EXT_framebuffer_sRGB
GL_EXT_gpu_program_parameters
GL_EXT_multi_draw_arrays
GL_EXT_packed_depth_stencil
GL_EXT_packed_float
GL_EXT_packed_pixels
GL_EXT_pixel_buffer_object
GL_EXT_point_parameters
GL_EXT_polygon_offset
GL_EXT_polygon_offset_clamp
GL_EXT_provoking_vertex
GL_EXT_rescale_normal
GL_EXT_secondary_color
GL_EXT_separate_specular_color
GL_EXT_shader_integer_mix
GL_EXT_shader_samples_identical
GL_EXT_shadow_funcs
GL_EXT_stencil_two_side
GL_EXT_stencil_wrap
GL_EXT_subtexture
GL_EXT_texture
GL_EXT_texture3D
GL_EXT_texture_array
GL_EXT_texture_compression_dxt1
GL_EXT_texture_compression_rgtc
GL_EXT_texture_compression_s3tc
GL_EXT_texture_cube_map
GL_EXT_texture_edge_clamp
GL_EXT_texture_env_add
GL_EXT_texture_env_combine
GL_EXT_texture_env_dot3
GL_EXT_texture_filter_anisotropic
GL_EXT_texture_integer
GL_EXT_texture_lod_bias
GL_EXT_texture_object
GL_EXT_texture_rectangle
GL_EXT_texture_sRGB
GL_EXT_texture_sRGB_decode
GL_EXT_texture_shared_exponent
GL_EXT_texture_snorm
GL_EXT_texture_swizzle
GL_EXT_timer_query
GL_EXT_transform_feedback
GL_EXT_vertex_array
GL_EXT_vertex_array_bgra
GL_IBM_multimode_draw_arrays
GL_IBM_rasterpos_clip
GL_IBM_texture_mirrored_repeat
GL_INGR_blend_func_separate
GL_INTEL_performance_query
GL_KHR_context_flush_control
GL_KHR_debug
GL_MESA_pack_invert
GL_MESA_texture_signed_rgba
GL_MESA_window_pos
GL_NV_blend_square
GL_NV_conditional_render
GL_NV_depth_clamp
GL_NV_light_max_exponent
GL_NV_packed_depth_stencil
GL_NV_primitive_restart
GL_NV_texgen_reflection
GL_NV_texture_barrier
GL_NV_texture_env_combine4
GL_NV_texture_rectangle
GL_OES_EGL_image
GL_OES_read_format
GL_S3_s3tc
GL_SGIS_generate_mipmap
GL_SGIS_texture_border_clamp
GL_SGIS_texture_edge_clamp
GL_SGIS_texture_lod
GL_SUN_multi_draw_arrays
```

**Comment by lewisjb** [comment]
I also get this problem: https://gist.github.com/lewisjb/551d60b5dfdf585d0fccd0a0a96522ce

**Comment by donbright** [comment]
i agree, what @nairainer and @lewisjb are describing sounds like what i encountered in 1785

video here

https://www.youtube.com/watch?v=CYO-nqs6oiQ

**Comment by nairaner** [comment]
Fixed with OpenCSG 1.4.2

**Comment by karlkashofer** [comment]
This bug is present in Debian Jessie and Stretch, a update is in the works via -backports, and here is a workaround:
http://karl.kashofer.org/?p=339
